bench-tps/src/main.rs

```diff
@@ -77,6 +77,8 @@ fn main() {
         println!("adding more lamports {}", extra);
         fund_keys(&clients[0], &id, &keypairs, extra);
     }
+    // `generate_keypairs` generates more keys than we care about. filter down to tx_count * 2 for correct spends later
+    keypairs.truncate(2 * tx_count);
```

Comment:
Why not just generate that many to begin with? Also, the above logic with looks at the last keypair and that keypair may be truncated here, so it doesn't seem right.

Reply:
It does this trick thing to parallelize the funding of the tx_count * 2 accounts. We create enough sacrificial accounts to allow us to "fanout" the clients balance into all the accounts tx_count * 2 in parallel. 
Hard to explain without a diagram but we those "extra" keypairs are left empty and so we shouldn't be dealing with them once funding is done.

Reply:
ok. maybe keypair0_balance should be then `keypairs[tx_count * 2 - 1]` and update the comment to say last keypair and update this comment to say 'generate_keypairs' generates extra keys to be able to have size-aligned funding batches for `fund_keys`.